timemachine/cpp/src/nonbonded_interaction_group.cu

```diff
@@ -220,11 +237,16 @@ void NonbondedInteractionGroup<RealType>::execute_device(
 
     const int tpb = warp_size;
     const int B = ceil_divide(N_, tpb);
+    const int K_B = ceil_divide(K_, tpb);
 
     // (ytz) see if we need to rebuild the neighborlist.
-    k_check_rebuild_coords_and_box<RealType>
-        <<<B, tpb, 0, stream>>>(N_, d_x, d_nblist_x_, d_box, d_nblist_box_, nblist_padding_, d_rebuild_nblist_);
 
+    // TODO: Is this more efficient as a single call (K_, d_atom_idxs_, ...)?
```

Comment:
Seems like it might be possible to avoid the overhead of an extra kernel launch by making `d_row_indices` and `d_col_indices` contiguous. But from what I understand the overhead is pretty small (~20 us if I'm reading [this](https://developer.nvidia.com/blog/understanding-the-visualization-of-overhead-and-latency-in-nsight-systems/) correctly), so it may not be worth it. @badisa or @proteneer might know better.

Reply:
Would be nice to avoid this, though as @mcwitt said its not a big deal.

We could put the row indices at the start of an array and the column in the rest of array and use NC/NR to point at them specifically. Since the number of total indices can't exceed N. Then you could have a single call here. Would require re-working more of the potential.

Though not sure I entirely see a case where these two kernel calls produces significantly fewer NBlist rebuilds, so using ` k_check_rebuild_coords_and_box`  might also be reasonable.